### PR TITLE
Fix warnings on disable USETCP_WIN macro

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Description
-----------
This change fixes warnings which appear on disabling the macro ipconfigUSE_TCP_WIN
This PR is a fix for the [customer reported issue](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/issues/930)

Test Steps
-----------
Run the [Echo Posix Demo](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix) with warnings enabled , the demo runs successfully without these warnings

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_TCP_WIN.c [hunk 1]
```diff
@@ -383,28 +385,30 @@
...
+            listLIST_ITEM_CONTAINER( pxNewListItem ) = ( struct xLIST * configLIST_VOLATILE ) pxList;
 
-        ( pxList->uxNumberOfItems )++;
-    }
+            ( pxList->uxNumberOfItems )++;
+        }
+    #endif /* if ( ipconfigUSE_TCP_WIN == 1 ) */
 /*-----------------------------------------------------------*/
 
     #if ( ipconfigUSE_TCP_WIN == 1 )
```
Comment (moninom1) [inline source/FreeRTOS_TCP_WIN.c:414]: There is already a flag, can we add above function to the same flag. Also is the complete file not expected to be within the ipconfigUSE_TCP_WIN flag.

File: source/FreeRTOS_ND.c [hunk 2]
```diff
@@ -479,14 +479,14 @@
         void FreeRTOS_PrintNDCache( void )
         {
             BaseType_t x, xCount = 0;
+            char pcBuffer[ 40 ];
```
Comment (moninom1) [inline source/FreeRTOS_ND.c:482]: What does this change, i do not see any flag difference between the 2 locations.

**Review by moninom1 — APPROVED** [review]

File: source/FreeRTOS_TCP_WIN.c (hunk 1 above)
Comment (moninom1) [reply source/FreeRTOS_TCP_WIN.c:414]: This will require separate effort, may be we can add it to backlog. Approving for now.

**Review by shubnil — APPROVED** [review]

**Pull request merged**